document null wrt pattern matching

IanKemp opened the pull request:
also fix typo (exp => expr)

IanKemp committed:
document null wrt pattern matching

also fix typo (exp => expr)

BillWagner commented on docs/csharp/language-reference/keywords/is.md:
```diff
@@ -79,7 +79,9 @@ The `is` expression is `true` if any of the following is true:
 - *expr* is an instance of a type that implements the *type* interface.
 
-If *exp* is `true` and `is` is used with an `if` statement, *varname* is assigned and has local scope within the `if` statement only.
+- *expr* does not evaluate to `null`.
```
I'll suggest moving this to Line 72, above.  Then, Line 72 would read

> The `is` expression is `true` if *expr* does not evaluate to `null`, and any of the following is true:

(By putting "*expr* does not evaluate to `null`" as one of the conditions in the list, *any* non-null object would mean `is` evaluates to `true`)

BillWagner reviewed (changes requested):
Thanks for adding this clarifying remark @IanKemp  IT was not clear that the expression cannot be null for the `is` expression to be `true`. I did have a suggestion for a different location for the remark. Can you make that change, and then I'll :shipit:.

Thanks for helping us improve docs.

BillWagner committed:
lift condition on expr being null

The condition "expr is not null" should not be part of the "any of the following" condition.
